packages/pyright-internal/src/languageServerBase.ts

```diff
@@ -1441,6 +1441,17 @@ export abstract class LanguageServerBase implements LanguageServerInterface, Dis
         };
     }
 
+    protected sendDiagnostics(params: PublishDiagnosticsParams[]) {
+        for (const param of params) {
+            if (param.diagnostics.length === 0) {
```

Comment:
This looks wrong. It's important for zero-length diganostic lists to be reported to the client. This is how old diagnostics are cleared for that file.

Reply:
this still send diagnostics. this logic is to clean up `this.documentsWithDiagnsotics` map we maintain. we use this info so that when workspace is closed, we clean up diagnostics reported if there is any.

Reply:
by the way, actual change is just renaming `_sendDiagnostics` to `sendDiagnostics`. actual logic didn't change.